client/restclient/restclient.go

```diff
@@ -1,36 +1,43 @@
 package restclient
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strings"
 	"time"
 
 	"github.com/pkg/errors"
 )
 
-// RestClient is the struct used interact with the Banyan REST API.
-type RestClient struct {
+// Client is the struct used interact with the Banyan REST API.
+type Client struct {
 	accessToken string
 	hostUrl     string
 	httpClient  *http.Client
 }
 
-const defaultHostUrl = "http://net.banyanops.com"
+const defaultHostUrl = "https://net.banyanops.com"
 
 // New creates a new client that will let the user interact with the REST API server.
-// As part of this it exchanges the given refreshtoken for an acesstoken.
-func New(hostUrl string, refreshToken string, apiToken string) (client *RestClient, err error) {
+// As part of this it exchanges the given refreshtoken or api key
+func New(hostUrl string, refreshToken string, apiToken string) (client *Client, err error) {
 	clientHostUrl := defaultHostUrl
 	if hostUrl != "" {
 		clientHostUrl = hostUrl
 	}
 
+	if !strings.HasSuffix(hostUrl, "/") {
```

Comment:
What's the difference between this `Client` type and the one above in this PR?

Reply:
apparently nothing. I removed it